### PR TITLE
feat: allow guest access to zone details

### PR DESCRIPTION
**Pull request by stakach**
and zone metadata, guests must supply a name param

### CODE REVIEW

File: src/placeos-rest-api/controllers/metadata.cr [hunk 1]
```diff
@@ -10,15 +10,24 @@ module PlaceOS::Api
     before_action :check_support, only: [:update, :update_alt, :destroy]
     before_action :find_zone, only: [:children]
 
+    # Allow guest access to system details of a single room
```
Comment (caspiano) [inline src/placeos-rest-api/controllers/metadata.cr:13]: Can this be rewritten more generally like this?

```suggestion
    # Allow unscoped read access to metadata
```

File: src/placeos-rest-api/controllers/metadata.cr [hunk 2]
```diff
@@ -10,15 +10,24 @@ module PlaceOS::Api
...
+    skip_action :check_oauth_scope, only: [:show, :children_metadata]
+
     getter zone : Model::Zone?
 
     # Fetch metadata for a model
     #
     # Filter for a specific metadata by name via `name` param
     def show
-      parent_id = params["id"]
+      zone_id = params["id"]
```
Comment (caspiano) [inline src/placeos-rest-api/controllers/metadata.cr:22]: This controller is meant to be generic to control systems and zones, hence the `parent_id`.
I believe `children` is specific to a `Zone` as there's no concept of hierarchy with `System`s

File: src/placeos-rest-api/controllers/metadata.cr [hunk 3]
```diff
@@ -106,5 +120,10 @@ module PlaceOS::Api
       # Find will raise a 404 (not found) if there is an error
       @zone = Model::Zone.find!(id)
     end
+
+    def guest_zones
```
Comment (caspiano) [inline src/placeos-rest-api/controllers/metadata.cr:124]: this method is `guest_zones` yet it's querying for `ControlSystem`s

File: src/placeos-rest-api/controllers/systems.cr [hunk 4]
```diff
@@ -27,6 +27,9 @@ module PlaceOS::Api
 
     before_action :ensure_json, only: [:create, :update, :update_alt, :execute]
 
+    # Allow guest access to system details of a single room
```
Comment (caspiano) [inline src/placeos-rest-api/controllers/systems.cr:30]:
```suggestion
    # Allow unscoped access to details of a single `ControlSystem`
```

File: src/placeos-rest-api/controllers/metadata.cr (hunk 3 above)
Comment (stakach) [reply src/placeos-rest-api/controllers/metadata.cr:124]: yeah as they only have access to the zone_id that are in that system

File: src/placeos-rest-api/controllers/metadata.cr (hunk 2 above)
Comment (stakach) [reply src/placeos-rest-api/controllers/metadata.cr:22]: oh right, forgot about that - will update this

File: src/placeos-rest-api/controllers/metadata.cr (hunk 3 above)
Comment (caspiano) [reply src/placeos-rest-api/controllers/metadata.cr:124]: ahh i see

Comment (caspiano) [reply src/placeos-rest-api/controllers/metadata.cr]: Can we assume that the `sys_id` will always be the last value in `roles`?

File: src/placeos-rest-api/controllers/metadata.cr [hunk 5]
```diff
@@ -106,5 +120,10 @@ module PlaceOS::Api
       # Find will raise a 404 (not found) if there is an error
       @zone = Model::Zone.find!(id)
     end
+
```
Comment (caspiano) [inline src/placeos-rest-api/controllers/metadata.cr:123]:
```suggestion
    # Fetch zones for system the current user has a role for
```

File: src/placeos-rest-api/controllers/metadata.cr (hunk 3 above)
Comment (stakach) [reply src/placeos-rest-api/controllers/metadata.cr]: yeah the guest JWT is defined here: https://docs.google.com/document/d/1LyHi1oeuIDE2WIdAmACpNaLhsITB6aALYk52EYVWAVc/edit

**Pull request merged**